Do-It/Controller/CreateDoItTableViewController.swift

```diff
@@ -162,3 +179,12 @@ extension CreateDoItTableViewController: CourseTableViewControllerDelegate {
         tableView.reloadRows(at: [IndexPath(row: Row.course.rawValue, section: 0)], with: .automatic)
     }
 }
+
+extension CreateDoItTableViewController: AlertDateTableViewControllerDelegate {
+    func alertDateTableViewController(_ alertVC: AlertDateTableViewController,
+                                      didSelectDate alertWhen: DateComponents, alertString: String) {
+        self.alertString = alertString
+        self.alertOption = Calendar.current.date(byAdding: alertWhen, to: self.date)!
```

Comment:
If the user changes the due date of the do-it after setting the alert time, the saved alert time has been invalidated. Store the DateComponents offset that describes the alert selection, and update it whenever the user changes the date.